x-pack/solutions/observability/plugins/synthetics/server/cases/suggestion.ts

```diff
@@ -0,0 +1,253 @@
+          );
+          const configIdsFilter = getSavedObjectKqlFilter({
+            field: 'config_id',
+            values: monitors.map((mon) => mon.latest_run.hits.hits[0]._source.config_id),
+          });
+
+          const resultMonitorsConfigWithLegacy =
+            await monitorConfigRepository.find<EncryptedSyntheticsMonitorAttributes>({
+              filter: configIdsFilter,
+              perPage: 10,
+            });
+
+          if (resultMonitorsConfigWithLegacy.total === 0) {
+            logger.error(`No Synthetics SavedObjects found for the related tests runs`);
+            return { suggestions: [] };
+          }
+
+          for (const savedObj of resultMonitorsConfigWithLegacy.saved_objects) {
+            if (!savedObj.error && savedObj.id && savedObj.attributes) {
+              savedObjectsAttrHash[savedObj.id] = savedObj.attributes;
+            }
+          }
+
+          const monitorsOverviewMetaData: Array<
+            OverviewStatusMetaData & { service: { name: string } }
+          > = monitors.flatMap((monitor) => {
+            return monitor.latest_run.hits.hits.flatMap((hit) => {
+              const source = hit._source;
+              const relatedSavedObjectAttr: EncryptedSyntheticsMonitorAttributes =
+                savedObjectsAttrHash[source.config_id];
+
+              if (!relatedSavedObjectAttr) {
+                logger.debug(
+                  `No related saved object found for config_id ${source.config_id} - possible deleted config, old run etc.`
+                );
+                return [];
+              }
+
+              return [
+                {
+                  monitorQueryId: source.monitor.id,
+                  configId: source.config_id,
+                  service: { name: source.service.name },
+                  status: source.monitor.status ?? 'unknown',
+                  name: source.monitor.name,
+                  isEnabled: relatedSavedObjectAttr.enabled,
+                  isStatusAlertEnabled: relatedSavedObjectAttr.alert?.status?.enabled ?? false,
+                  type: source.monitor.type,
+                  schedule: relatedSavedObjectAttr.schedule.number,
+                  tags: relatedSavedObjectAttr.tags ?? [],
+                  maintenanceWindows: relatedSavedObjectAttr.maintenance_windows ?? [],
+                  timestamp: source['@timestamp'],
+                  spaces: source.meta.space_id,
+                  locationLabel: source.observer.geo.name,
+                  locationId: source.observer.name,
+                  urls: source.url.full,
+                  projectId: relatedSavedObjectAttr.project_id,
+                },
+              ];
+            });
+          });
+
+          const suggestions = monitorsOverviewMetaData.flatMap((monitor) => {
+            const url = locator?.getRedirectUrl({
+              configId: monitor.configId,
+              locationId: monitor.locationId,
+              spaceId: monitor.spaces ?? [],
```

Comment:
This is not generating the correct url when clicking on a url generated for a monitor in a non default space. Let's use the `spaceId` that is passed to the handler.

```suggestion
              spaceId,
```

Reply:
@cesco-f, fixed!